yt_dlp/extractor/echo360.py

```diff
@@ -0,0 +1,108 @@
+                'ext': 'mp4',
+                'title': 'EXSC634_Online_Workshop_Week_4.mp4',
+                'duration': 6659.72,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+    ]
+
+    def _call_api(self, host, video_id, media_id, session_token, **kwargs):
+        return self._download_json(
+            f'https://{host}/api/ui/echoplayer/public-links/{video_id}/media/{media_id}/player-properties',
+            video_id, headers={'Authorization': f'Bearer {session_token}'}, **kwargs)
+
+    def _replace_url_query(self, url, query_string):
+        if query_string is not None:
+            return urlunparse(urlparse(url)._replace(query=query_string))
+        return url
+
+    def _get_query_string(self, uri, query_strings):
+        uri_base = urlparse(uri)._replace(query='', fragment='').geturl()
+        for query_string in query_strings:
+            try:
+                if re.match(query_string['uriPattern'], uri_base):
+                    return query_string['queryString']
+            except re.error as re_error:
+                self.report_warning(f'Error in query string pattern `{re_error.pattern}`: {re_error.msg}')
+        return None
+
+    def _parse_mediapackage(self, video):
+        video_id = video['playableAudioVideo']['mediaId']
+        query_strings = traverse_obj(video, ('sourceQueryStrings', 'queryStrings')) or []
+
+        formats = []
+        for track in traverse_obj(video, ('playableAudioVideo', 'playableMedias', ...)):
+            href = track.get('uri')
+            if href is None:
+                continue
+            href = self._replace_url_query(href, self._get_query_string(href, query_strings))
+            if track.get('isHls') or determine_ext(href, None) == 'm3u8':
+                hls_formats = self._extract_m3u8_formats(
+                    href, video_id, live=track.get('isLive'), m3u8_id='hls',
+                    entry_protocol='m3u8_native', fatal=False
+                )
+
+                for hls_format in hls_formats:
+                    query_string = self._get_query_string(hls_format['url'], query_strings)
+                    hls_format['extra_param_to_segment_url'] = query_string
+                    hls_format['url'] = self._replace_url_query(hls_format['url'], query_string)
+
+                formats.extend(hls_formats)
+
+        return {
+            'id': video_id,
+            'formats': formats,
+            'title': video.get('mediaName'),
+            'duration': float_or_none(self._search_regex(
+                r'PT([\d.]+)S', traverse_obj(video, ('playableAudioVideo', 'duration')),
+                'video duration', fatal=False)),
+        }
+
+    def _real_extract(self, url):
+        host, video_id = self._match_valid_url(url).group('host', 'id')
+        webpage = self._download_webpage(url, video_id)
+
+        player_config = self._search_json(
+            r'Echo\["mediaPlayerBootstrapApp"\]\("', webpage, 'player config',
```

Comment:
Some pages now use `echoPlayerV2FullBootstrapApp` instead - this fixes the issue I was having!

```suggestion
            r'Echo\["(mediaPlayerBootstrapApp|echoPlayerV2FullBootstrapApp)"\]\("', webpage, 'player config',
```

Reply:
```suggestion
            r'Echo\["(?:mediaPlayer|echoPlayerV2Full)BootstrapApp"\]\("', webpage, 'player config',
```